api/views/course_admin.py

```diff
@@ -16,11 +17,13 @@ class CourseAdminViewSet(viewsets.GenericViewSet):
     permission_classes = [TeacherAccessPermission, ]
     queryset = CanvasCourse.objects.all()
     serializer_class = CourseSerializer
+    search_fields = ['user__first_name', 'user__last_name', 'user__email']
+    filter_backends = [filters.SearchFilter, DjangoFilterBackend]
 
     @action(detail=True, methods=['get'], url_path='registered-users')
     def registered_users(self, request, pk=None):
         course = get_object_or_404(self.get_queryset(), pk=pk)
-        course_registration_list = course.canvascourseregistration_set.all()
+        course_registration_list = self.filter_queryset(course.canvascourseregistration_set.all())
```

Comment:
Here you need to manually filter the course_registration set. If you don't know how to do it ask Abi to help you.